### PR TITLE
Fixed the hard-coded Dune 2000 production tab hotkeys

### PR DESCRIPTION
**Pull request by Mailaender**
A followup of https://github.com/OpenRA/OpenRA/pull/7384. This makes them configurable in the Input tab and displays them in the tooltip.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs [hunk 1]
```diff
@@ -422,7 +422,8 @@ Action InitInputPanel(Widget panel)
 					{ "ProductionTypeInfantryKey", "Infantry Tab" },
 					{ "ProductionTypeVehicleKey", "Vehicle Tab" },
 					{ "ProductionTypeAircraftKey", "Aircraft Tab" },
-					{ "ProductionTypeNavalKey", "Naval Tab" }
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs]: Why did you remove the ```Naval Tab``` completely here?

Comment (Mailaender) [inline OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs]: Woops. Fixed. Thanks.

File: OpenRA.Game/Settings.cs [hunk 2]
```diff
@@ -206,6 +206,8 @@ public class KeySettings
 		public Hotkey ProductionTypeVehicleKey = new Hotkey(Keycode.Y, Modifiers.None);
 		public Hotkey ProductionTypeAircraftKey = new Hotkey(Keycode.U, Modifiers.None);
 		public Hotkey ProductionTypeNavalKey = new Hotkey(Keycode.I, Modifiers.None);
+		public Hotkey ProductionTypeTankKey = new Hotkey(Keycode.O, Modifiers.None);
+		public Hotkey ProductionTypeMerchantKey = new Hotkey(Keycode.P, Modifiers.None);
```
Comment (obrakmann) [inline OpenRA.Game/Settings.cs]: `P` conflicts with the Diplomacy dialog.

**Comment by obrakmann** [comment]
I think the different order and missing production types show that the initial implementation isn't as flexible as it should be to suit all mods.

I think ideally the buttons should move down the row of buttons as you progress from left to right on the keyboard. However, in d2k, there's a key that does nothing (`R` for the Defense queue), another key makes you jump back up the row (the Tank queue (`O`) comes before the Aircraft queue (`U`)). 

I think a generally more flexible approach would be to assign slots to each of the buttons and then  bind the keys not to the type of queue but to the slot it occupies. That would solve both the gaps and the jumping back and forth.

**Comment by pchote** [comment]
>  think the different order and missing production types show that the initial implementation isn't as flexible as it should be to suit all mods.

I think that's being a bit too polite - our hotkey implementation is inflexible and a bit crap in general.  We are well past the point where hardcoded hotkey types no longer work well.

We should move towards a system where ButtonWidgets can bind to named hotkey types that are defined in an external yaml file and customized per-mod in settings.yaml.

File: OpenRA.Game/Settings.cs (hunk 2 above)
Comment (Mailaender) [inline OpenRA.Game/Settings.cs]: Switched.

**Comment by obrakmann** [comment]
For the time being, this fixes the hard-coded shortcuts and works as expected. :+1:

**Comment by abcdefg30** [comment]
Except
merchanttab
(it looks unpolished), this looks good to me.

**Comment by penev92** [comment]
"Starport Tab:" might just fit ;)
Also I like the name better.

**Comment by penev92** [comment]
As much as this is better than bleed, the keys are still crap.
R does nothing, infantry are on T. While this may be OK for other mods, it's not for D2k.
And after that the tabs go: Vehicles, Tanks, Aircraft, Starport; but the hotkeys go Vehicles, Aircraft, Tanks, Starport.
I get why all this is happening, but I'd rather have them non-configurable, if that means we can hardcode the proper keys.

**Comment by Mailaender** [comment]
> "Starport Tab:" might just fit ;)

Updated.

**Comment by penev92** [comment]
Also we could just make the tab order match the key order with E (for buildings) as a starting key.
Then match second tab to R, third to T, etc.
I don't see why production tabs need to be configurable. (although it might be just me...)

**Comment by TodorBalabanov** [comment]
Pavel, I would like to ask you something in private. Please, write me an email: todor.balabanov@gmail.com

**Comment by Mailaender** [comment]
> I don't see why production tabs need to be configurable. (although it might be just me...)

Because not everyone has a keyboard layout like you and the hotkey tooltip is tied to it.

**Comment by penev92** [comment]
The current (RA-styled) hotkeys make D2k rather frustrating for the first several games (until you manage to get used to the odd layout), but as it seems this isn't going anywhere I'm going to merge it now and hope we can do proper hotkeys per-mod sooner rather than later.

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/d828e827de9d451de4073ee4f63c352061e88b94...57bdf95d9c599102b3d0f6bf83d30040b2bfcd4d)
